### PR TITLE
[MRG+1] Made Python wrappers for public Cython functions

### PR DESCRIPTION
**Pull request by mdmueller**
This is meant to address #1323. This is mostly just copy-pasting, so let me know if the wrappers should have some other format.

### CODE REVIEW

**Comment by ahojnnes** [comment]
Looks good, thanks! Let's wait for CI.

**Comment by blink1073** [comment]
@ahojnnes, what do you think about a generic test to make sure all of our imported `*` functions have a docstring?

**Comment by ahojnnes** [comment]
@blink1073 What do you mean by "`*` imported functions"?

**Comment by mdmueller** [comment]
Ah, it looks like there are problems running the examples in the docstrings. Is there a way to import functions needed for these examples? `ellipse_perimeter` (in `skimage.draw`) seems to be the culprit.

File: skimage/draw/_draw.pyx [hunk 1]
```diff
@@ -486,7 +486,7 @@ def circle_perimeter_aa(Py_ssize_t cy, Py_ssize_t cx, Py_ssize_t radius,
             np.array(val, dtype=np.float))
 
 
-def ellipse_perimeter(Py_ssize_t cy, Py_ssize_t cx, Py_ssize_t yradius,
+def _ellipse_perimeter(Py_ssize_t cy, Py_ssize_t cx, Py_ssize_t yradius,
                       Py_ssize_t xradius, double orientation=0, shape=None):
```
Comment (stefanv) [inline skimage/draw/_draw.pyx]: Remember to indent these lines too

**Comment by mdmueller** [comment]
That's odd. The example at fault is this:
```
from skimage.draw import ellipse_perimeter
img = np.zeros((25, 25), dtype=np.uint8)
rr, cc = ellipse_perimeter(10, 10, 6, 8)
img[cc, rr] = 1
result = hough_ellipse(img, threshold=8)
print(result)
```
When I run this, I get `[(10, 10.0, 10.0, 8.0, 6.0, 0.0)]` while the docstring says it should be `[(10, 10.0, 8.0, 6.0, 0.0, 10.0)]`. I don't know anything about Hough transforms, so is this a problem with the docstring example or an actual bug?

**Comment by JDWarner** [comment]
@mdmueller I can confirm your result. I'm running a very recent build of master, too. Anaconda, OSX 10.6.8:

``` python
from skimage.draw import ellipse_perimeter
from skimage.transform import hough_ellipse
img = np.zeros((25, 25), dtype=np.uint8)
rr, cc = ellipse_perimeter(10, 10, 6, 8)
img[cc, rr] = 1
result = hough_ellipse(img, threshold=8)
print(result)
```
yields the sorted result
`[(10, 10.0, 10.0, 8.0, 6.0, 0.0)]`

Were these doctests not being run because they were in Cython files? That used to be a thing, but [checking out their Wiki](https://github.com/cython/cython/wiki/FAQ) supposedly "new versions of Cython" should work with doctests. I'm not sure when that got added to be able to compare against the version of Cython we test with.

Also worth noting we have some improvements to the Hough transforms in the wings in #1421. Whichever of these gets merged first is going to create a fun rebase for the other... unless we revert the Hough changes and re-implement those after #1421 is sorted.

**Comment by mdmueller** [comment]
So should we wait here for #1421?

**Comment by sciunto** [comment]
#1421 is merged. What's the status?

**Comment by ahojnnes** [comment]
LGTM, but needs a rebase.

File: skimage/feature/corner.py [hunk 2]
```diff
@@ -853,3 +853,123 @@ def corner_peaks(image, min_distance=10, threshold_abs=0, threshold_rel=0.1,
...
+    """Compute Moravec corner measure response image.
+
+    This is one of the simplest corner detectors and is comparatively fast but
+    has several limitations (e.g. not rotation invariant).
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    window_size : int, optional (default 1)
```
Comment (sciunto) [inline skimage/feature/corner.py]: remove `(default 1)`

File: skimage/segmentation/quickshift_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    """Segments image using quickshift clustering in Color-(x,y) space.
+
+    Produces an oversegmentation of the image using the quickshift mode-seeking
+    algorithm.
+
+    Parameters
+    ----------
+    image : (width, height, channels) ndarray
+        Input image.
+    ratio : float, optional, between 0 and 1 (default 1).
```
Comment (sciunto) [inline skimage/segmentation/quickshift_wrapper.py]: remove default here and below, as well. We do not write default values for python functions.

File: skimage/segmentation/quickshift_wrapper.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        Higher means fewer clusters.
+    return_tree : bool, optional (default False)
+        Whether to return the full segmentation hierarchy tree and distances.
+    sigma : float, optional (default 0)
+        Width for Gaussian smoothing as preprocessing. Zero means no smoothing.
+    convert2lab : bool, optional (default True)
+        Whether the input should be converted to Lab colorspace prior to
+        segmentation. For this purpose, the input is assumed to be RGB.
+    random_seed : None (default) or int, optional
+        Random seed used for breaking ties.
```
Comment (sciunto) [inline skimage/segmentation/quickshift_wrapper.py]: Here say "If `None`, then ..."

File: skimage/transform/hough_transform.py [hunk 5]
```diff
@@ -151,3 +152,144 @@ def hough_circle(image, radius, normalize=True, full_output=False):
...
                          normalize=normalize, full_output=full_output)
+
+def hough_ellipse(img, threshold=4, accuracy=1, min_size=4, max_size=None):
+    """Perform an elliptical Hough transform.
+
+    Parameters
+    ----------
+    img : (M, N) ndarray
+        Input image with nonzero values representing edges.
+    threshold: int, optional (default 4)
```
Comment (sciunto) [inline skimage/transform/hough_transform.py]: remove default here as well.

File: skimage/transform/hough_transform.py [hunk 6]
```diff
@@ -151,3 +152,144 @@ def hough_circle(image, radius, normalize=True, full_output=False):
...
+        image dimension.
+
+    Returns
+    -------
+    result : ndarray with fields [(accumulator, y0, x0, a, b, orientation)]
+          Where ``(yc, xc)`` is the center, ``(a, b)`` the major and minor
+          axes, respectively. The `orientation` value follows
+          `skimage.draw.ellipse_perimeter` convention.
+
+    Notes
```
Comment (sciunto) [inline skimage/transform/hough_transform.py]: There is two notes sections. This one must be removed.

File: skimage/transform/hough_transform.py [hunk 7]
```diff
@@ -151,3 +152,144 @@ def hough_circle(image, radius, normalize=True, full_output=False):
...
+    return _hough_ellipse(img, threshold, accuracy, min_size, max_size)
+
+def hough_line(img, theta=None):
+    """Perform a straight line Hough transform.
+
+    Parameters
+    ----------
+    img : (M, N) ndarray
+        Input image with nonzero values representing edges.
+    theta : 1D ndarray of double
```
Comment (sciunto) [inline skimage/transform/hough_transform.py]: `, optional`

File: skimage/transform/hough_transform.py [hunk 8]
```diff
@@ -151,3 +152,144 @@ def hough_circle(image, radius, normalize=True, full_output=False):
...
+
+def probabilistic_hough_line(img, threshold=10, line_length=50,
+                             line_gap=10, theta=None):
+    """Return lines from a progressive probabilistic line Hough transform.
+
+    Parameters
+    ----------
+    img : (M, N) ndarray
+        Input image with nonzero values representing edges.
+    threshold : int, optional (default 10)
```
Comment (sciunto) [inline skimage/transform/hough_transform.py]: Remove default.

File: skimage/transform/hough_transform.py [hunk 9]
```diff
@@ -151,3 +152,144 @@ def hough_circle(image, radius, normalize=True, full_output=False):
...
+        Threshold
+    line_length : int, optional (default 50)
+        Minimum accepted length of detected lines.
+        Increase the parameter to extract longer lines.
+    line_gap : int, optional, (default 10)
+        Maximum gap between pixels to still form a line.
+        Increase the parameter to merge broken lines more aggresively.
+    theta : 1D ndarray, dtype=double, optional, default (-pi/2 .. pi/2)
+        Angles at which to compute the transform, in radians.
```
Comment (sciunto) [inline skimage/transform/hough_transform.py]: If None...

**Comment by sciunto** [comment]
I made few comments on the docstring style.

@mdmueller Do you have some time to fix the docstring and rebase, please?

**Comment by mdmueller** [comment]
@sciunto I fixed the docstring and tried a rebase, though I'm not sure if I handled merge conflicts correctly since I don't remember much about the code. If I made any mistakes, let me know and I can correct them.

File: skimage/draw/_draw.pyx [hunk 10]
```diff
@@ -766,14 +766,14 @@ def _bezier_segment(Py_ssize_t y0, Py_ssize_t x0,
...
+    rr, cc = _line(x0, y0, x2, y2)
     px.extend(rr)
     py.extend(cc)
 
     return np.array(py, dtype=np.intp), np.array(px, dtype=np.intp)
 
 
-def bezier_curve(Py_ssize_t y0, Py_ssize_t x0,
+def _bezier_curve(Py_ssize_t y0, Py_ssize_t x0,
                  Py_ssize_t y1, Py_ssize_t x1,
```
Comment (ahojnnes) [inline skimage/draw/_draw.pyx]: Increase indentation by 1 here and in the next two lines.

File: skimage/draw/draw.py [hunk 11]
```diff
@@ -272,3 +272,367 @@ def set_color(img, coords, color, alpha=1):
...
+           [0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0, 1, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0, 0, 1, 0, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 1, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 1, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
+
+    """
+    return _line(y, x, y2, x2)
+
```
Comment (ahojnnes) [inline skimage/draw/draw.py]: Add one extra empty line between functions.

File: skimage/draw/draw.py [hunk 12]
```diff
@@ -272,3 +272,367 @@ def set_color(img, coords, color, alpha=1):
...
+           [  0,   0,  56, 255,  56,   0,   0,   0,   0,   0],
+           [  0,   0,   0,  56, 255,  56,   0,   0,   0,   0],
+           [  0,   0,   0,   0,  56, 255,  56,   0,   0,   0],
+           [  0,   0,   0,   0,   0,  56, 255,  56,   0,   0],
+           [  0,   0,   0,   0,   0,   0,  56, 255,  56,   0],
+           [  0,   0,   0,   0,   0,   0,   0,  56, 255,   0],
+           [  0,   0,   0,   0,   0,   0,   0,   0,   0,   0]], dtype=uint8)
+    """
+    return _line_aa(y1, x1, y2, x2)
+
```
Comment (ahojnnes) [inline skimage/draw/draw.py]: Here also another empty line, also elsewhere.

File: skimage/measure/pnpoly.py [hunk 13]
```diff
@@ -0,0 +1,59 @@
+from ._pnpoly import _grid_points_in_poly, _points_in_poly
+
```
Comment (ahojnnes) [inline skimage/measure/pnpoly.py]: +1 empty line

File: skimage/feature/corner.py [hunk 14]
```diff
@@ -841,3 +841,123 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=0.1,
...
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0],
+           [0, 0, 1, 1, 1, 0, 0],
+           [0, 0, 1, 2, 1, 0, 0],
+           [0, 0, 1, 1, 1, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0]])
+    """
+    return _corner_moravec(image, window_size)
+
```
Comment (ahojnnes) [inline skimage/feature/corner.py]: +1 empty line

File: skimage/feature/corner.py [hunk 15]
```diff
@@ -841,3 +841,123 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=0.1,
         return np.transpose(peaks.nonzero())
     else:
         return peaks
+
```
Comment (ahojnnes) [inline skimage/feature/corner.py]: +1 empty line

File: skimage/measure/pnpoly.py [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+    This function is a wrapper for Cython code.
+
+    Returns
+    -------
+    mask : (M, N) ndarray of bool
+        True where the grid falls inside the polygon.
+
+    """
+    return _grid_points_in_poly(shape, verts)
+
```
Comment (ahojnnes) [inline skimage/measure/pnpoly.py]: +1 empty line

File: skimage/segmentation/quickshift_wrapper.py [hunk 17]
```diff
@@ -0,0 +1,56 @@
+from ._quickshift import quickshift as _quickshift
+
```
Comment (ahojnnes) [inline skimage/segmentation/quickshift_wrapper.py]: +1 empty line

File: skimage/transform/hough_transform.py [hunk 18]
```diff
@@ -306,26 +180,113 @@ def hough_ellipse(img, threshold=4, accuracy=1, min_size=4, max_size=None):
...
+
+    Apply the Hough transform:
+
+    >>> out, angles, d = hough_line(img)
+
+    .. plot:: hough_tf.py
+
+    """
+    return _hough_line(img, theta)
+
```
Comment (ahojnnes) [inline skimage/transform/hough_transform.py]: +1 empty line

**Comment by ahojnnes** [comment]
@mdmueller Thanks for the quick response. Did you check that the doc-strings are up to date in your new Python wrapper functions?

File: skimage/draw/_draw.pyx [hunk 19]
```diff
@@ -38,7 +38,7 @@ def _coords_inside_image(rr, cc, shape, val=None):
         return rr[mask], cc[mask], val[mask]
 
 
-def line(Py_ssize_t y0, Py_ssize_t x0, Py_ssize_t y1, Py_ssize_t x1):
+def _line(Py_ssize_t y0, Py_ssize_t x0, Py_ssize_t y2, Py_ssize_t x2):
```
Comment (sciunto) [inline skimage/draw/_draw.pyx]: why x2, y2?

Comment (sciunto) [inline skimage/draw/_draw.pyx]: and not x1, y1?

File: skimage/draw/draw.py [hunk 20]
```diff
@@ -272,3 +272,367 @@ def set_color(img, coords, color, alpha=1):
     vals = img[rr, cc] * (1 - alpha)
 
     img[rr, cc] = vals + color
+
+
+def line(y, x, y2, x2):
```
Comment (sciunto) [inline skimage/draw/draw.py]: looks weird to me. Either x0, y0 and x1, y1 OR x1, y1 and x2, y2

File: skimage/draw/draw.py [hunk 21]
```diff
@@ -272,3 +272,367 @@ def set_color(img, coords, color, alpha=1):
...
+           [0, 0, 0, 1, 1, 1, 0, 0, 0, 0],
+           [0, 0, 0, 1, 1, 1, 0, 0, 0, 0],
+           [0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
+
+    """
+    return _polygon(y, x, shape)
+
```
Comment (sciunto) [inline skimage/draw/draw.py]: One empty line here as well.

File: skimage/draw/draw.py [hunk 22]
```diff
@@ -272,3 +272,367 @@ def set_color(img, coords, color, alpha=1):
...
+           [  0, 211,  43,   0,   0,   0,  43, 211,   0,   0],
+           [  0, 255,   0,   0,   0,   0,   0, 255,   0,   0],
+           [  0, 211,  43,   0,   0,   0,  43, 211,   0,   0],
+           [  0,  60, 194,  43,   0,  43, 194,  60,   0,   0],
+           [  0,   0,  60, 211, 255, 211,  60,   0,   0,   0],
+           [  0,   0,   0,   0,   0,   0,   0,   0,   0,   0],
+           [  0,   0,   0,   0,   0,   0,   0,   0,   0,   0]], dtype=uint8)
+    """
+    return _circle_perimeter_aa(cy, cx, radius, shape)
+
```
Comment (sciunto) [inline skimage/draw/draw.py]: + 1 empyt line.

File: skimage/draw/draw.py [hunk 23]
```diff
@@ -272,3 +272,367 @@ def set_color(img, coords, color, alpha=1):
...
+def ellipse_perimeter(cy, cx, yradius, xradius, orientation=0, shape=None):
+    """Generate ellipse perimeter coordinates.
+
+    Parameters
+    ----------
+    cy, cx : int
+        Centre coordinate of ellipse.
+    yradius, xradius : int
+        Minor and major semi-axes. ``(x/xradius)**2 + (y/yradius)**2 = 1``.
+    orientation : double, optional (default 0)
```
Comment (sciunto) [inline skimage/draw/draw.py]: You can remove `(default 0)`

File: skimage/draw/draw.py [hunk 24]
```diff
@@ -272,3 +272,367 @@ def set_color(img, coords, color, alpha=1):
...
+        Major axis orientation in clockwise direction as radians.
+    shape : tuple, optional
+        Image shape which is used to determine the maximum extent of output pixel
+        coordinates. This is useful for ellipses which exceed the image size.
+        By default the full extent of the ellipse are used.
```
Comment (sciunto) [inline skimage/draw/draw.py]: If `None`, the full extent of the ellipse is used.

File: skimage/draw/draw.py [hunk 25]
```diff
@@ -272,3 +272,367 @@ def set_color(img, coords, color, alpha=1):
...
+           [0, 1, 0, 0, 0, 0, 0, 0, 0, 1],
+           [0, 1, 0, 0, 0, 0, 0, 0, 0, 1],
+           [0, 1, 0, 0, 0, 0, 0, 0, 0, 1],
+           [0, 0, 1, 0, 0, 0, 0, 0, 1, 0],
+           [0, 0, 0, 1, 1, 1, 1, 1, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
+
+    """
+    return _ellipse_perimeter(cy, cx, yradius, xradius, orientation, shape)
+
```
Comment (sciunto) [inline skimage/draw/draw.py]: +1 empty line...

File: skimage/draw/draw.py [hunk 26]
```diff
@@ -272,3 +272,367 @@ def set_color(img, coords, color, alpha=1):
...
+    y1, x1 : int
+        Coordinates of the middle control point.
+    y2, x2 : int
+        Coordinates of the last control point.
+    weight : double
+        Middle control point weight, it describes the line tension.
+    shape : tuple, optional
+        Image shape which is used to determine the maximum extent of output
+        pixel coordinates. This is useful for curves which exceed the image
+        size. By default the full extent of the curve are used.
```
Comment (sciunto) [inline skimage/draw/draw.py]: If `None`, the full extent of the ellipse is used.

File: skimage/feature/corner.py [hunk 27]
```diff
@@ -841,3 +841,123 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=0.1,
...
+    This is one of the simplest corner detectors and is comparatively fast but
+    has several limitations (e.g. not rotation invariant).
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    window_size : int, optional
+        Window size.
```
Comment (sciunto) [inline skimage/feature/corner.py]: Can you give a hint on how to choose this value?

Comment (ahojnnes) [inline skimage/feature/corner.py]: @sciunto This shouldn't be part of this PR...

Comment (sciunto) [inline skimage/feature/corner.py]: If you consider this as out of the scope of this PR, it's OK.

Comment (sciunto) [inline skimage/feature/corner.py]: @ahojnnes Yes, I was on my run :D

File: skimage/transform/hough_transform.py [hunk 28]
```diff
@@ -1,74 +1,13 @@
...
+from scipy import ndimage
+from .. import measure, morphology
+from ._hough_transform import (_hough_circle, _hough_ellipse,
+                               _hough_line, _probabilistic_hough_line)
 
 
 def hough_line_peaks(hspace, angles, dists, min_distance=9, min_angle=10,
                      threshold=None, num_peaks=np.inf):
-    """Return peaks in a straight line Hough transform.
+    """Return peaks in hough transform.
```
Comment (sciunto) [inline skimage/transform/hough_transform.py]: Can you keep the upper  case (Hough), please? it's a name.

File: skimage/feature/corner.py (hunk 27 above)
Comment (ahojnnes) [inline skimage/feature/corner.py]: @sciunto No problem, just trying to reduce the work load for @mdmueller.

File: skimage/transform/hough_transform.py [hunk 29]
```diff
@@ -306,26 +180,113 @@ def hough_ellipse(img, threshold=4, accuracy=1, min_size=4, max_size=None):
...
+        Input image with nonzero values representing edges.
+    theta : 1D ndarray of double
+        Angles at which to compute the transform, in radians.
+        Defaults to -pi/2 .. pi/2
+
+    Returns
+    -------
+    H : 2-D ndarray of uint64
+        Hough transform accumulator.
+    theta : ndarray, optional
```
Comment (sciunto) [inline skimage/transform/hough_transform.py]: How this output can be optional?

**Comment by sciunto** [comment]
Very nice piece of work!

File: skimage/feature/corner.py [hunk 30]
```diff
@@ -841,3 +841,123 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=0.1,
...
+    This function is a wrapper for Cython code.
+
+    References
+    ----------
+    .. [1] http://kiwi.cs.dal.ca/~dparks/CornerDetection/moravec.htm
+    .. [2] http://en.wikipedia.org/wiki/Corner_detection
+
+    Examples
+    --------
+    >>> from skimage.feature import corner_moravec
```
Comment (sciunto) [inline skimage/feature/corner.py]: You can delete this line.

**Comment by sciunto** [comment]
Once done, it should be ready to go.

File: skimage/transform/hough_transform.py (hunk 29 above)
Comment (mdmueller) [inline skimage/transform/hough_transform.py]: It defaults to a range from -pi/2 to pi/2

**Comment by mdmueller** [comment]
Ok I just made a new commit addressing some of the mentioned issues.

File: skimage/transform/hough_transform.py (hunk 29 above)
Comment (sciunto) [inline skimage/transform/hough_transform.py]: But that should be for the parameters section, not returns, right?

**Comment by sciunto** [comment]
There's also a problem with the tests. Let us know if you need help. I can make a PR against your branch tonight if that helps.

**Comment by mdmueller** [comment]
Yeah that would be great, could you make a PR against this branch? I'm not sure if I remember the code well enough to diagnose this stuff at the moment. Thanks!

**Comment by soupault** [comment]
@mdmueller you can grant us permissions to work on your branch now -> https://github.com/blog/2247-improving-collaboration-with-forks.

**Comment by mdmueller** [comment]
Ok, just enabled that option

**Comment by amccaugh** [comment]
Any word on this PR?  (I have a PR in the pipeline which depends on it)

**Comment by sciunto** [comment]
I didn't have time to look at it and I'm quite busy these days. Anyone else?

File: skimage/draw/draw.py (hunk 12 above)
Comment (sciunto) [inline skimage/draw/draw.py]: done already

File: skimage/measure/pnpoly.py (hunk 13 above)
Comment (sciunto) [inline skimage/measure/pnpoly.py]: done.

File: skimage/feature/corner.py (hunk 15 above)
Comment (sciunto) [inline skimage/feature/corner.py]: done

File: skimage/measure/pnpoly.py (hunk 16 above)
Comment (sciunto) [inline skimage/measure/pnpoly.py]: done

File: skimage/segmentation/quickshift_wrapper.py (hunk 17 above)
Comment (sciunto) [inline skimage/segmentation/quickshift_wrapper.py]: done.

File: skimage/transform/hough_transform.py (hunk 18 above)
Comment (sciunto) [inline skimage/transform/hough_transform.py]: done.

File: skimage/draw/draw.py (hunk 21 above)
Comment (sciunto) [inline skimage/draw/draw.py]: done.

File: skimage/draw/draw.py (hunk 22 above)
Comment (sciunto) [inline skimage/draw/draw.py]: done.

File: skimage/draw/draw.py (hunk 25 above)
Comment (sciunto) [inline skimage/draw/draw.py]: done.

File: skimage/transform/hough_transform.py (hunk 29 above)
Comment (sciunto) [inline skimage/transform/hough_transform.py]: fixed.

**Comment by sciunto** [comment]
Replaced by the PR mentioned above. I'm closing this one. Thanks @mdmueller
